shell/models/__tests__/node.ts

```diff
@@ -0,0 +1,52 @@
+import Node from '@shell/models/management.cattle.io.node';
+
+describe('class MgmtNode', () => {
```

Comment:
As far as I can see, there's a lot of duplicated logic. Not entirely sure how this is handled in our OOP architecture, but I guess we may have a subclass to be extended for both the models and have 1 test suite there?

Reply:
Duplicated logic isn't a major issue here in this unit test land where we want our tests to be isolated and deterministic. Centralizing logic in tests can save you a lot of work when you've got confounding variables in your test suite but doing so needlessly can lead to mistakes such as an error in the central logic causing all tests to fail without indicating why in an obvious way.